doc/design.md

```diff
@@ -1,51 +1,203 @@
 # Design
 
-## Project Structure Definition
-`dsn~project-structure-definition~1`
+## Validators
 
-The project structure is defined by a file structure in the resources folder of this plugin.
-There are two folders: `require_exist` and `require_exact`. 
-For files in the `require_exist` folder, this plugin only checks if they exist.
-If they don't exist it creates the file with the content from the file in `require_exist` as a template.
+`dsn~validators~1`
 
-For the files in the `require_exact` this plugin also validates that the content is exactly the same.
+As an interface for the different validations we created the `Validator` interface. It only contains the `validate` method:
 
-Rationale:
-This way of defining the project structure makes it very convenient to add new content.
-Now knowledge about the internal implementation of this plugin is required.
-That allows users to quickly add or modify the template.
+```java
+public List<ValidationFinding> validate();
+```
+
+The `ValidationFinding` consists of:
+
+* A message
+* Optionally: A fix for the finding (function closure that changes the project so that the finding will not happen again)
```

Comment:
```suggestion
* An optional fix for the violation finding
```

A fix is a function closure that changes the project so that the fixed finding does not happen again.

Reply:
```suggestion
* Optionally: A fix for the finding (function closure that changes the project so that the finding will not happen again)

A fix is a function closure that changes the project so that the fixed finding does not happen again.
```